### PR TITLE
Curator inbox add events if not in database

### PR DESCRIPTION
**Pull request by jgoley**
Allows users to add an event and associate source/IRs.
[PT Story](https://www.pivotaltracker.com/story/show/141958319)

### CODE REVIEW

File: methods/articles.coffee [hunk 1]
```diff
@@ -4,20 +4,17 @@ Meteor.methods
...
-          userEventId: source.userEventId
-          url: source.url
-          deleted:
-            $in: [null, false]
-        ).fetch()
-        unless existingArticle.length is 0
-          throw new Meteor.Error(501, 'This article has already been added')
-        else
+      # Check if Article is in collection
+      source = Articles.findOne(url: $regex: new RegExp("#{sourceId}$"))
```
Comment (nathanathan) [inline methods/articles.coffee]: This could match the wrong article if one has an id that is a substring of another. For example, `10` and `110`.

**Comment by nathanathan** [comment]
Only showing the create event button when there are no search results could cause it to go unnoticed. For instance, if someone searches for a disease like Dengue, they might see a couple matches for previous years or other countries and see that there are no events that fit their article yet have no indication they can add a new event.

I got this server error when I tried to create an event and associate IRs with it:
```
I20170329-02:15:25.276(-4)? Exception while invoking method 'addIncidentsToEvent' TypeError: Cannot read property 'title' of undefined
I20170329-02:15:25.277(-4)?     at [object Object].Meteor.methods.addIncidentsToEvent (methods/incidentReports.coffee:73:22)
I20170329-02:15:25.278(-4)?     at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1711:12)
I20170329-02:15:25.278(-4)?     at packages/ddp-server/livedata_server.js:711:19
I20170329-02:15:25.278(-4)?     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
I20170329-02:15:25.279(-4)?     at packages/ddp-server/livedata_server.js:709:40
I20170329-02:15:25.279(-4)?     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
I20170329-02:15:25.280(-4)?     at packages/ddp-server/livedata_server.js:707:46
I20170329-02:15:25.280(-4)?     at Session.method (packages/ddp-server/livedata_server.js:681:23)
I20170329-02:15:25.280(-4)?     at packages/ddp-server/livedata_server.js:551:43
```

**Comment by jgoley** [comment]
@nathanathan I think I've fixed all the issues.
- Updated the Article query to: `url: $regex: "#{regexEscape(source.url)}$"`
- Show the 'add new event' button in the select2 list always
- And fixed that error when associating IRs (my bad)

**Comment by nathanathan** [comment]
When I tried to create an event via the "Add Source to Event" button, I got this error:
```
Uncaught TypeError: Cannot read property 'fetch' of undefined
    at Object.submit #createEvent (app.js:6316)
    at blaze.js:3718
    at Function.Template._withTemplateInstanceFunc (blaze.js:3687)
    at Blaze.View.<anonymous> (blaze.js:3717)
    at blaze.js:2560
    at Object.Blaze._withCurrentView (blaze.js:2214)
    at Blaze._DOMRange.<anonymous> (blaze.js:2559)
    at HTMLFormElement.<anonymous> (blaze.js:835)
    at HTMLBodyElement.dispatch (jquery.js:4723)
    at HTMLBodyElement.elemData.handle (jquery.js:4391)
```

**Comment by nathanathan** [comment]
👍 Looks good

**Pull request merged**